modules/form-field/react/lib/Label.tsx

```diff
@@ -16,19 +17,32 @@ const labelStyles = [
     padding: 0,
   },
   (props: LabelProps) => {
-    if (props.labelPosition === FormFieldLabelPosition.Left) {
-      return {
-        display: 'inline-block',
-        verticalAlign: 'top',
-        marginTop: 10, // Input height - font line height / 2
-        marginRight: spacing.l,
-        width: 180,
-      };
-    }
-
     return {
-      display: 'block',
-      marginBottom: spacing.xxxs,
+      ...(props.labelPosition === FormFieldLabelPosition.Left
+        ? {
+            display: 'inline-block',
+            verticalAlign: 'top',
+            marginTop: 10, // Input height - font line height / 2
+            marginRight: spacing.l,
+            width: 180,
+          }
+        : {
+            display: 'block',
+            marginBottom: spacing.xxxs,
+          }),
+      ...(props.required
+        ? {
+            ':after': {
+              content: '"*"',
+              color: inputColors.error.border,
```

Comment:
Nit: Do you mind using `cinnamon500` instead? We try to avoid using the semantic colors for things they're not intended for.